core/services/ocr2/validate/validate.go

```diff
@@ -123,6 +125,41 @@ func validateSpec(tree *toml.Tree, spec job.Job) error {
 	return nil
 }
 
+type coreConfig struct {
```

Comment:
Is this type just for testing ATM?
I'm hoping that `Command` won't actually end up here, and that plugin versioning isn't tangled up at this level just yet.

Reply:
@jmank88 Do you mean that it won't bend up in the job spec long-term or just in this package? This was how I intended it to be implemented initially, but I can see a case for some kind of indirection (i.e. a var defined here and then the actual binary path pulled from eg. static config)

Reply:
Although we will eventually need to be able to configure a particular version of a plugin for a job, I don't think that embedding literal paths to binaries will be a good UX, and that seems to expose implementation details that we will want to be able to change later (e.g. with nix).

No matter which way we implement it, we should try to have reasonable defaults to support minimal config. In this case, we could infer from plugin name that we are looking for `chainlink-<pluginname>`. We happen to know exactly what those are in the current state, but in the future we could go looking for new things dynamically. Maybe we assume `smartcontract/chainlink-<pluginname>/cmd/chainlink-<pluginname>`, but they have the option to provide any full module path. As for version, we could configure global versions at the root/core level, and optionally override them here, with another field, or with a `@<semver>` suffix.

Reply:
Oh and this highlights the asymmetry of median. We are currently on track for `smartcontractkit/chainlink-feeds/cmd/chainlink-median`, but does that actually make sense? If in the future someone wants to use mode, average, or any kind of other aggregate reduction on a totally different type of data, would we really want separate plugin binaries for each? Or would we just prefer one `chainlink-feeds` plugin that can be configured to use median, mode, avg, etc.?

Reply:
FWIW I think that most of the current median code is written as "big.Int feed" code, with median playing a very small part.

Reply:
assuming that selecting the median is really just a small part of the plugin (I think it is), then I think it's better to make the "aggregation method" configurable, so in effect we would have a "feeds" plugin with aggregation_method="median" or similar

Reply:
Is this the kind of thing you had in mind @jmank88 ? https://github.com/smartcontractkit/chainlink/pull/10665/commits/56ebd8f13d529d693dd509abfd6a4a2ee35a5f6a

Note I'm not going further here since there is no version (and in fact no "repository" abstraction which we discussed separately -- in my mind this can be added later when implemented)

Reply:
Yeah, although even simpler ATM since we install the binaries on the system path currently, so just the short form `chainlink-*` would work.